docs/reference/deployment-targets-yaml-reference.md

```diff
@@ -0,0 +1,271 @@
+one of the following values: `SYNCING`, `USABLE`, `UNUSABLE`, or `CORDONED`. It should
+not be modified.
+
+#### state_message
+
+The `state_message` property provides an additional message associated with
+the deployment target's state. It should not be modified.
+
+#### subject_rules
+
+The `subject_rules` property holds a list of subject rules that determine who
+can trigger promotions when the deployment target is associated with them. Each
+rule must include the `type`. Depending on the rule `type`, the `subject_id`
+or `git_login` may also be required.
+
+##### type
+
+The `type` property of a subject rule can have the following values:
+`ANY`, `USER`, `ROLE`, `GROUP`, or `AUTO`. If `ANY` is used, any user can trigger
+the promotion. If `USER` is used, only specified users (who must be assigned to
+the project) can trigger a promotion. For `ROLE`, only specified user roles defined
+within the project can trigger a promotion.
+
+##### subject_id
+
+The `subject_id` property is required for `USER`, `ROLE`, and `GROUP` rule types.
+For `USER`, the `subject_id` contains the UUID of the user. If the rule type is `ROLE`,
+it should be the name of the project role (e.g., `Admin`, `Contributor`).
+
+##### git_login
+
+The `git_login` property is used only for `USER` rule types when specifying a user
+by their git handle (e.g., GitHub login) instead of their UUID.
+
+#### object_rules
+
+The `object_rules` property contains a list of object rules that define which branches,
+tags, or pull requests can trigger promotions. Each rule must include the `type`,
+`match_mode`, and `pattern` properties.
+
+##### type
+
+The `type` property of an object rule can have the following values: `BRANCH`, `TAG`,
+or `PR`.
+
+##### match_mode
+
+The `match_mode` property determines how the pattern matches the name of the git
+reference (branch, tag, or pull request). The `match_mode` can be one of the following:
+`ALL`, `EXACT`, or `REGEX`.
+
+##### pattern
+
+The `pattern` property is a string used to match the name of the git reference specified
+by the rule type (branch, tag, or pull request). The pattern is only used for `EXACT`
+and `REGEX` match modes. In `EXACT` mode, an exact string comparison is performed between
+the branch/tag name and the pattern. In `REGEX` mode, the pattern is treated as a regular
+expression to match the reference name. The regular expressions are Perl-compatible, and
+you can find more information about the syntax in
+the [the Erlang re module documentation](https://www.erlang.org/doc/man/re.html)
+
+#### active
+
+The `active` property is a boolean value indicating whether the deployment target is active
+or inactive. It should not be modified.
+
+#### bookmark_parameter1, bookmark_parameter2, bookmark_parameter3
+
+The `bookmark_parameter1`, `bookmark_parameter2` and `bookmark_parameter3` properties are
+string values that can be used to index future deployments.
```

Comment:
We should add more data about how this links deployment targets and parameterized promotions.
Change it to something similar to this:

The `bookmark_parameter1`, `bookmark_parameter2` and `bookmark_parameter3` properties are
string values that represent the names of the promotion parameters. You can later use values of those parameters to filter deployments in deployment history.
For example, if you create a parameterized promotion that has parameter `environment` that can have values `staging` and `production` and connect it to a deployment target that has `bookmark_parameter1` set to `environment` it will later allow you to filter deployment history to show only promotions where e.g. `staging` was passed as the value of parameter.